test/generators/csharp/CSharpGenerator.spec.ts

```diff
@@ -0,0 +1,155 @@
+import { CSharpGenerator } from '../../../src/generators';
+
+describe('CSharpGenerator', () => {
+  let generator: CSharpGenerator;
+  beforeEach(() => {
+    generator = new CSharpGenerator();
+  });
+
+  test('should not render reserved keyword', async () => {
+    const doc = {
+      $id: 'Address',
+      type: 'object',
+      properties: {
+        enum: { type: 'string' },
+        reservedEnum: { type: 'string' }
+      },
+      additionalProperties: false
+    };
+
+    const inputModel = await generator.process(doc);
+    const model = inputModel.models['Address'];
+
+    let classModel = await generator.renderClass(model, inputModel);
+    expect(classModel.result).toMatchSnapshot();
+
+    classModel = await generator.render(model, inputModel);
+    expect(classModel.result).toMatchSnapshot();
+  });
+  test('should render `class` type', async () => {
+    const doc = {
+      $id: '_address',
+      type: 'object',
+      properties: {
+        street_name: { type: 'string' },
+        city: { type: 'string', description: 'City description' },
+        state: { type: 'string' },
+        house_number: { type: 'number' },
+        marriage: { type: 'boolean', description: 'Status if marriage live in given house' },
+        members: { oneOf: [{ type: 'string' }, { type: 'number' }, { type: 'boolean' }], },
+        tuple_type: { type: 'array', items: [{ type: 'string' }, { type: 'number' }] },
+        array_type: { type: 'array', items: { type: 'string' } },
+      },
+      required: ['street_name', 'city', 'state', 'house_number', 'array_type'],
+      additionalProperties: {
+        type: 'string'
+      },
+      patternProperties: {
+        '^S(.?*)test&': {
+          type: 'string'
+        }
+      },
+    };
+
+    const inputModel = await generator.process(doc);
+    const model = inputModel.models['_address'];
+
+    let classModel = await generator.renderClass(model, inputModel);
+    expect(classModel.result).toMatchSnapshot();
```

Comment:
Please aware that using snapshot isn't a good idea, because then you must "believe" of content saved in snapshot. Why did you start using the snapshots?

Reply:
Because it is simply too much work to continue to manually update the expected results when there is this much content. To me, this is not a problem, as you inspect the changes of the snapshots in PR's and manually when they are changed.

Reply:
Personally I don't like snapshot, but if you want use them, feel free.

Reply:
If it was as simple as copy/paste the content, I would mind. But the content it produces need to be converted, because of different escaped characters, which is making it a pain 😆